Classes/Neos/Neos/Ui/Fusion/Helper/NodeInfoHelper.php

```diff
@@ -74,7 +74,13 @@ public function renderNode(NodeInterface $node, ControllerContext $controllerCon
             $nodeInfo['uri'] = $this->uri($node, $controllerContext);
         }
 
-        foreach ($node->getChildNodes() as $childNode) {
+        // child nodes for document tree, respecting the `baseNodeType` filter
+        $documentChildNodes = $node->getChildNodes($this->baseNodeType);
+        // child nodes for content tree, must not include those nodes filtered out by `baseNodeType`
+        $contentChildNodes = $node->getChildNodes('!Neos.Neos:Document');
```

Comment:
Sure, I had my mind on cleaning this up, since we have this kind of mess in other lines of this file already (and all over the system in general). I think I'll go with the second option you suggested since this logic is pretty specific to the tree.